app/services/publish_interactor.rb

```diff
@@ -32,23 +24,32 @@ def call(params)
 
   private
 
-  def fetch_or_create_organization_publisher(organization, publisher)
-    organization_publisher = OrganizationPublisher
-                             .where(organization: organization)
-                             .where(publisher: publisher)
-                             .first
+  def authorized_to_publish?(organization, publisher)
+    authorized = OrganizationPublisher
+                 .where(organization: organization)
+                 .where(publisher: publisher)
+                 .exists?
 
     # if the publisher is already authorized to publish on behalf of this
-    # organization, great, use that OrganizationPublisher record
-    return organization_publisher if organization_publisher
+    # organization, great
+    return true if authorized
 
     # if not, and the publisher is not a super publisher, bail
-    return nil unless publisher.super_publisher
+    raise MR::NotAuthorizedToPublish unless publisher.super_publisher
```

Comment:
I like the idea of using custom exceptions when needed, specially if we were to use similar checks on other points of the API and return the same msg/status (on a grape rescue_from block). 
But not using exceptions for control flow like here.
this can just be an `if`:
```
unless publisher.super_publisher
  @error = [NOT_AUTHORIZED_ERR, 401]  # NOT_AUTHORIZED_ERR can be a class const (frozen str) 
  return false
end

OrganizationPublisher.create(organization: organization, publisher: publisher)

```
Also, it's slower:
```
b = Benchmark.bm do |x|
  x.report { puts 'yey' if true }
  x.report { raise 'msg' rescue puts 'yey' }
end
puts b
  0.000000   0.000000   0.000000 (  0.000012)
  0.000000   0.000000   0.000000 (  0.000065)
=> nil
```

Reply:
Yes, the idea was to start using this elsewhere too so I'm not always creating the same [message, code] array all over the place. But yeah, it's only in one place right now so I'll revert to using a simple conditional until this is actually needed.